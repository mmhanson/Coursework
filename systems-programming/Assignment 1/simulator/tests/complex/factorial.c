int fact(int f)
{
  if(f == 2)
    return 2;
  return f * fact(f-1);
}

int main()
{
  int input;
  scanf(&input);
  printf("%d\n", fact(input0));
}
