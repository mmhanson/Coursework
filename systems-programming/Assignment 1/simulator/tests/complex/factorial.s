main:
	subl	$8, %esp
	readr	%edi
	call	fact
	printr	%eax
	addl	$8, %esp
	ret
fact:
	pushl	%ebx
	movl	%edi, %ebx
	movl	$2, %r8d
	cmpl	%r8d, %edi
	je	.L3
	subl	$1, %edi
	call	fact
	imull	%ebx, %eax
	jmp	.L2
.L3:
	movl	$2, %eax
.L2:
	popl	%ebx
	ret
