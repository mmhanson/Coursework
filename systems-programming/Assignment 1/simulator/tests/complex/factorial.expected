120 (0x78)
